fendermint/vm/interpreter/src/fvm/gas/mod.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use fendermint_actor_gas::Gas;
+
+pub(crate) mod default;
+
+/// Handles the gas modeling in the current blockchain
+pub trait GasLayer {
```

Comment:
- Let's call this `GasMarket`.
- Let's document that during normal operation, it's supposed to be backed by an actor.
- A more future-proof, pure interface for this:

```rust
pub trait GasMarket<State, Input> {
    /// Fetches the current state of the gas market.
    fn current() -> State;
    /// Updates the state of the gas market from the aggregated gas utilization stats for the block.
    fn update(Input) -> State;
}

struct State {
    block_gas_limit: Gas,
    /// This is the base_fee, but termed in a more abstract way so our terminology is not overfitted to EIP-1559.
    burn_fee_per_gas: TokenAmount,
}
```

Reply:
Sounds good, then the actor is really simple, just writing the state and reading the state.